src/Errata/Internal/Render.hs

```diff
@@ -36,50 +38,110 @@ renderErrors :: Source source => source -> [Errata] -> TB.Builder
 renderErrors source errs = unsplit "\n\n" prettified
     where
         sortedErrata = sortOn (blockLocation . errataBlock) errs
-        slines = let xs = sourceToLines source in A.listArray (0, length xs - 1) xs
+        slines = sourceToLines source
         prettified = map (renderErrata slines) sortedErrata
 
 -- | A single pretty error from metadata and source lines.
-renderErrata :: Source source => A.Array Int source -> Errata -> TB.Builder
+renderErrata :: Source source => [source] -> Errata -> TB.Builder
 renderErrata slines (Errata {..}) = errorMessage
     where
+        -- | Header block + body blocks
+        blocks = errataBlock : errataBlocks
+
+        -- The pointers grouped by line.
+        blockPointersGrouped = I.fromListWith (<>) . map (pointerLine &&& pure) . blockPointers <$> blocks
```

Comment:
I don't think the `Control.Arrow` import is justified for just this one use of `(&&&)` :P